src/canvasController.ts

```diff
@@ -0,0 +1,91 @@
+import { CanvasValues } from './constants';
+import { Enemy } from './enemy';
+import { Player } from './player';
+import { Weapon } from './weapon';
+
+export class CanvasController {
+  private ctx: CanvasRenderingContext2D;
+  constructor(ctx: CanvasRenderingContext2D) {
+    this.ctx = ctx;
+  }
+
+  private drawPlayer(player: Player) {
+    this.ctx.save();
+    this.ctx.translate(player.body.x, player.body.y);
+    this.ctx.rotate(player.body.heading);
+
+    this.ctx.fillStyle = player.color;
+    this.ctx.fillRect(
+      (-1 * player.body.width) / 2,
+      (-1 * player.body.height) / 2,
+      player.body.width,
+      player.body.height
+    );
+
+    this.ctx.fillStyle = player.faceColor;
+    this.ctx.fillRect(
+      (-1 * player.body.width) / 2,
+      player.body.height / 2 - 2,
+      player.body.width,
+      2
+    );
+
+    if (player.isAttacking) {
+      const weaponRotation =
+        (player.weapon.swingAngle * (Date.now() - player.attackTimer)) /
+        player.weapon.attackDuration;
+      this.drawWeapon(player.weapon, weaponRotation, player.body.width / 2);
+    }
+
+    this.ctx.restore();
+  }
+
+  private drawEnemy(enemy: Enemy) {
+    this.ctx.fillStyle = 'red';
```

Comment:
Maybe this should be enemy.color? Come to think of it, why does this need to be different from drawPlayer? Maybe you should have a single drawCharacter function to handle both (by taking the more abstract Character type)?

Reply:
Done 3d0bd7b1afdf0ad0657e0b0a724012bf47aa1e73